paper/sections/materials.tex

```diff
@@ -9,16 +9,38 @@ \section{Materials and Methods}
 \subsection{Dataset}
 \label{sec:dataset}
 
-The SwissProt UniProt database consisting of $10,780$ proteins with rich sequence and substrate
-annotations~\cite{boeckmann2003swiss} was subsampled to include $780$ membrane transport proteins and $660$
-non-transport proteins. The transport proteins were divided into $7$ substrate-specific classes ($70$ amino acid
+The SwissProt UniProt database \sout{consisting of $10,780$ proteins} with rich sequence and substrate
+annotations~\cite{boeckmann2003swiss} was subsampled to include $780$ membrane transporter proteins and $660$
+non-transporter proteins. The transport proteins were divided into $7$ substrate-specific classes ($70$ amino acid
 transporters, $60$ anion transporters, $260$ cation transporters, $60$ electron transporters, $60$ sugar
 transporters, $70$ protein/mRNA transporters, and $200$ other transporters). With the addition of 600 non-transporter
 proteins, the total dataset contained $1,380$ protein sequences. \GK{Hamid: state explicitly the discrepancy between
 the 780 transporters we are using versus the 900 included in the original paper. This total appears to correspond to
 the ''main'' section of the dataset described in the original paper, which suggests we threw out the validation set?
 We should be clear about this}
 
+\Hamid{About the SwissProt, the initial $10,780$ sequences in the dataset, how they got to the final 
+dataset from there, and the subset we used for our experiment :: i wasn't sure how to mix it in with the text.
+[i can also add a table showing the number of proteins in each one]}
+{
+The dataset we used for this study is the same as the one being used in the initial work which is 
+available at \url{http://bioinfo.noble.org/TrSSP}. The authors in~\cite{mishra2014prediction} were initially collected 
+10,780 transporter, carrier, and channel proteins that were well characterized at the protein level and had clear substrate 
+annotations from the SwissProt(release 2013-03)~\cite{boeckmann2003swiss}. They then removed the transporters 
+with more than two substrate specificities, sequences with biological function annotations based solely on sequence 
+similarity, and sequences with greater than 70\% similarity. The final dataset contains 660 non-transporters and 
+900 transporters being divided into following 7 substrate classes: amino acid, anion, cation, electron, protein/mRNA, 
+sugar, and other.\\
+These $1,560$ sequences were then divided into two datasets: 1) Main dataset for the total of $1,380$ sequences 
+and 2) Independent Dataset which contains $180$ sequences. The work in~\cite{mishra2014prediction} reports on results of 
+all the developed models on the main dataset and the results from the best model (AAIndex+PSSM) on the independent set. 
+Through this study, we used the main dataset and experimented on all the models being developed on that set. 
```

Comment:
@HamidHeyde  - thanks for clarifying. While they may use the terms "main" and "independent," this is very standardly referred to as train/test set, and validation set. The validation set is deliberately left out of any exploration and is used once you've picked your model. So, while we're evaluating performance on our train/test sets, the fact that we never compare performance on the validation set to me feels like a large piece that we are missing: our models may have higher accuracy/F1/precision/recall/you-name-it, but they could also generalize more poorly out-of-sample; it is impossible to know the answer without using a validation set, as well.

Can you please create the independent dataset and evaluate our models (or at least the "best" one(s)) using it?

@glatard , what are your thoughts on this?

Reply:
Hey @gkiar, thanks for pointing this out! As discussed, I agree that we should evaluate our models on the independent set (180 sequences), in particular:
* Measure how far the model closest to the paper model is, in the independent set (replication aim)
* Measure what is the variability across all our models, in the independent set (methods flexibility aim)
@HamidHeyde, do you think you could do that?

Reply:
Hi @glatard, @gkiar.
Thanks for the review @gkiar,

So, if i'm right, you guys want me to try all the 156 models on the validation set and report the results? (Table 1,2,3 for validation set as well?)

Reply:
Yep, exactly! Just grab the extra 180 sequences from the database, matching the distributions in the original paper, and then evaluate **every** already-fit model on this new collection. :)

Reply:
Ok, it may take some time, but no problem 👍

Reply:
Perfect. I'll merge this PR and start reviewing your methods once the other (text based) edits are done, and we can merge the changes as a result of this afterwards. Sound good?

Reply:
Great 👍